lib/plug/cowboy/telemetry.ex

```diff
@@ -0,0 +1,75 @@
+defmodule Plug.Cowboy.Telemetry do
+  def init(stream_id, req, opts) do
+    system_time = System.system_time()
+    start_time = System.monotonic_time()
+
+    :telemetry.execute([:plug_cowboy, :stream_handler, :start], %{system_time: system_time}, %{
+      host: req.host,
+      method: req.method,
+      request_path: req.path,
+      port: req.port,
+      remote_ip: req.peer |> elem(0),
+      scheme: req.scheme,
+      req_headers: req.headers |> Map.to_list(),
```

Comment:
I matched the key names to Plug `conn` keys but if we make this a library that's cowboy focused, we might want to stick to the names that cowboy uses

Reply:
Agreed. :+1:

Reply:
Ideally we want to do as little processing here as possible.

Reply:
Oh, the best approach here is to just pass the whole `req` as metadata! :)

Reply:
I'm in progress re-writing this in erlang and that's the path I'm taking (ie: pretty much just forwarding the data as it's provided by cowboy)

I'll try to push that up soon

Reply:
Ok, so I pushed up a re-write of this, it's in `erlang`, uses cowboy terminology and avoids doing any pre-processing.

Open to another round of feedback on this!